ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/KafkaConfigStoreTest.java

```diff
@@ -0,0 +1,334 @@
+  }
+
+  @Test
+  public void shouldIgnoreRecordsWithDifferentKey() {
+    // Given:
+    addPollResult("foo", "val".getBytes());
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, serializer.serialize("", savedProperties));
+    expectRead(consumerBefore);
+
+    // When:
+    getKsqlConfig();
+
+    // Then:
+    verifyDrainLog(consumerBefore, 2);
+    inOrder.verifyNoMoreInteractions();
+  }
+
+  @Test
+  public void shouldIgnoreRecordsWithDifferentKeyWithinPoll() {
+    // Given:
+    addPollResult(
+        "foo", "val".getBytes(),
+        KafkaConfigStore.CONFIG_MSG_KEY, serializer.serialize("", savedProperties)
+    );
+    expectRead(consumerBefore);
+
+    // When:
+    getKsqlConfig();
+
+    // Then:
+    verifyDrainLog(consumerBefore, 1);
+    inOrder.verifyNoMoreInteractions();
+  }
+
+  @Test
+  public void shouldPollToEndOfTopic() {
+    // Given:
+    addPollResult("foo", "val".getBytes());
+    addPollResult("bar", "baz".getBytes());
+    expectRead(consumerBefore);
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, savedProperties);
+    expectRead(consumerAfter);
+
+    // When:
+    getKsqlConfig();
+
+    // Then:
+    verifyDrainLog(consumerBefore, 2);
+    verifyProduce();
+  }
+
+  @Test
+  public void shouldWriteConfigIfNoConfigWritten() {
+    // Given:
+    expectRead(consumerBefore);
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, properties);
+    expectRead(consumerAfter);
+
+    // When:
+    getKsqlConfig();
+
+    // Then:
+    verifyDrainLog(consumerBefore, 0);
+    verifyProduce();
+  }
+
+
+  @Test
+  public void shouldUseFirstPolledConfig() {
+    // Given:
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, savedProperties, badProperties);
+    expectRead(consumerBefore);
+
+    // When:
+    final KsqlConfig mergedConfig = getKsqlConfig();
+
+    // Then:
+    verifyMergedConfig(mergedConfig);
+  }
+
+  @Test
+  public void shouldNotWriteConfigIfExists() {
+    // Given:
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, savedProperties);
+    expectRead(consumerBefore);
+
+    // When:
+    getKsqlConfig();
+
+    // Then:
+    verifyNoMoreInteractions(producer);
+  }
+
+  @Test
+  public void shouldReadConfigAfterWrite() {
+    // Given:
+    expectRead(consumerBefore);
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, savedProperties, properties);
+    expectRead(consumerAfter);
+
+    // When:
+    final KsqlConfig mergedConfig = getKsqlConfig();
+
+    // Then:
+    verifyDrainLog(consumerBefore, 0);
+    verifyProduce();
+    verifyDrainLog(consumerAfter, 1);
+    verifyMergedConfig(mergedConfig);
+  }
+
+  @Test
+  public void shouldMergeExistingConfigIfExists() {
+    // Given:
+    addPollResult(KafkaConfigStore.CONFIG_MSG_KEY, savedProperties);
+    expectRead(consumerBefore);
+
+    // When:
+    final KsqlConfig mergedConfig = getKsqlConfig();
+
+    // Then:
+    verifyMergedConfig(mergedConfig);
+  }
+
+  @Test
+  public void shouldDeserializeEmptyContentsToNull() {
```

Comment:
name still seems out of alignment with test. Test looks to deserialise to empty map, not null.

```suggestion
  public void shouldDeserializeEmptyContentsToEmptyProps() {
```